cmd/agent-actions/actions/patch_node_handler.go

```diff
@@ -0,0 +1,29 @@
+package actions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"k8s.io/client-go/kubernetes"
+
+	"castai-agent/cmd/agent-actions/telemetry"
+)
+
+func newPatchNodeHandler(clientset *kubernetes.Clientset) ActionHandler {
+	return &patchNodeHandler{
+		clientset: clientset,
+	}
+}
+
+type patchNodeHandler struct {
+	clientset *kubernetes.Clientset
```

Comment:
Depend on interface.

Reply:
Changed, but this was left to be implemented in separate PR as this one already contains quite few changes.